editor/editor_file_dialog.cpp

```diff
@@ -86,7 +123,9 @@ void EditorFileDialog::_notification(int p_what) {
 					if (preview_wheel_index >= 8) {
 						preview_wheel_index = 0;
 					}
-					Ref<Texture2D> frame = item_list->get_theme_icon("Progress" + itos(preview_wheel_index + 1), SNAME("EditorIcons"));
+
+					Ref<Texture2D> progress[] = { theme_cache.progress_1, theme_cache.progress_2, theme_cache.progress_3, theme_cache.progress_4, theme_cache.progress_5, theme_cache.progress_6, theme_cache.progress_7, theme_cache.progress_8 };
```

Comment:
Constructing this array each time in process seems a bit wasteful. Maybe the theme cache could have the array directly instead of these 8 individual members.